src/evalfunc.c

```diff
@@ -9819,7 +9829,65 @@ f_search(typval_T *argvars, typval_T *rettv)
 {
     int		flags = 0;
 
-    rettv->vval.v_number = search_cmn(argvars, NULL, &flags);
+    rettv->vval.v_number = search_cmn(argvars, NULL, NULL, &flags);
+}
+
+/*
+ * "searchstr()" function
+ */
+    static void
+f_searchstr(typval_T *argvars, typval_T *rettv)
+{
+    pos_T	match_pos;
+    pos_T	end_pos;
+    linenr_T	start_lnum;
+    linenr_T	end_lnum;
+    colnr_T	start_col;
+    colnr_T	end_col;
+    char_u	buf[BUFSIZ];
+    int		multi_line = 0;
+    int		matchlen = 0;
+    int		flags = 0;
+
+    rettv->vval.v_string = NULL;
+    rettv->v_type = VAR_STRING;
+
+    rettv->vval.v_number = search_cmn(argvars, &match_pos, &end_pos, &flags);
```

Comment:
If invalid arguments are passed to the function, then search_cmn() will return FAIL.  You should handle that here.